File: app/src/modules/storage/enums/storage-provider-type.ts
export enum StorageProviderType {
    Assets = 'Assets',
    Attachments = 'Attachments',
    External = 'External',
}

export const StorageProviderTypeLabels: Record<StorageProviderType, string> = {
    [StorageProviderType.Assets]: 'Dokumente und Medien',
    [StorageProviderType.Attachments]: 'Prozessanlagen',
    [StorageProviderType.External]: 'Externe Dokumentenablage',
};

export const StorageProviderTypes = [
    StorageProviderType.Assets,
    StorageProviderType.Attachments,
    StorageProviderType.External,
];

File: backend/src/main/java/de/aivot/gover/backend/storage/enums/StorageProviderType.java
package de.aivot.gover.backend.storage.enums;

public enum StorageProviderType {
    Assets("Dokumente und Medien"),
    Attachments("Prozessanlagen"),
    External("Externe Dokumentenablage"),
    ;

    private final String label;

    StorageProviderType(String label) {
        this.label = label;
    }

    public String getLabel() {
        return label;
    }
}
